python/tests/test_dlis.py

```diff
@@ -185,3 +131,50 @@ def test_indexedobjects(f):
     assert len(f.comments)     == 1
     assert len(f.messages)     == 1
 
+def test_initial_load(fpath):
+    with dlisio.load(fpath) as (f, *tail):
+        # Only fileheader, origin, frame, and channel should be loaded
+        assert len(f.indexedobjects) == 4
+
+def test_objecttype_load(fpath):
+    with dlisio.load(fpath) as (f, *tail):
+        fp = core.fingerprint('PARAMETER', 'PARAM1', 10, 0)
+        parameters = f.parameters
+
+        assert len(parameters) == 3
+        assert fp in f.indexedobjects['PARAMETER']
+
+def test_loading_new_objecttype(fpath):
+    with dlisio.load(fpath) as (f, *tail):
+        fp = core.fingerprint('TOOL', 'TOOL1', 10, 0)
```

Comment:
Depending on the test intention it might be good to check first that `fp not in f.indexedobjects['TOOL']`
plus: you will make sure that this particular action will load the object
minus: maybe it shouldn't matter when the object is loaded as long as it's loaded when needed